The -Wl,-pie flag compiles successfully on Tiger but does not link.
So test linking instead.

--- config-scripts/cups-compiler.m4.orig
+++ config-scripts/cups-compiler.m4
@@ -156,7 +156,7 @@
 	OLDCFLAGS="$CFLAGS"
 	AS_CASE(["$host_os_name"], [darwin*], [
 	    CFLAGS="$CFLAGS -fPIE -Wl,-pie"
-	    AC_COMPILE_IFELSE([AC_LANG_PROGRAM()], [
+	    AC_LINK_IFELSE([AC_LANG_PROGRAM()], [
 		PIEFLAGS="-fPIE -Wl,-pie"
 		AC_MSG_RESULT([yes])
 	    ], [
@@ -164,7 +164,7 @@
 	    ])
 	], [*], [
 	    CFLAGS="$CFLAGS -fPIE -pie"
-	    AC_COMPILE_IFELSE([AC_LANG_PROGRAM()], [
+	    AC_LINK_IFELSE([AC_LANG_PROGRAM()], [
 		PIEFLAGS="-fPIE -pie"
 		AC_MSG_RESULT([yes])
 	    ], [
